### PR TITLE
terminus_font: make possible to use variants

### PR DESCRIPTION
**Pull request by andreoss**
- terminus-font: make variants customisable
- maintainers: add andreoss

###### Description of changes

the source code contains extra variants for several glyphs
this change makes them available for customization 

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/data/fonts/terminus-font/default.nix [hunk 1]
```diff
@@ -21,6 +21,10 @@ stdenv.mkDerivation rec {
   postPatch = ''
     substituteInPlace Makefile --replace 'fc-cache' '#fc-cache'
     substituteInPlace Makefile --replace 'gzip'     'gzip -n'
```
Comment (SuperSandro2000) [inline pkgs/data/fonts/terminus-font/default.nix:23]:
```suggestion
    substituteInPlace Makefile \
      --replace 'fc-cache' '#fc-cache' \
      --replace 'gzip' 'gzip -n'
```

File: pkgs/data/fonts/terminus-font/default.nix [hunk 2]
```diff
@@ -1,7 +1,7 @@
 { lib, stdenv, fetchurl, python3
 , bdftopcf, mkfontscale
+, variants ? []
```
Comment (SuperSandro2000) [inline pkgs/data/fonts/terminus-font/default.nix:3]: Can we make this an enum style input for all the variants listed here https://github.com/bachan/terminus-font-vietnamese/tree/master/alt ?

Comment (andreoss) [reply pkgs/data/fonts/terminus-font/default.nix:3]: @SuperSandro2000 Is there a good example how to use `types.enum` at this level? Enums don't seem to be a documented feature of Nix language and I don't see that enums were used anywhere in pkgs/

Comment (SuperSandro2000) [reply pkgs/data/fonts/terminus-font/default.nix:4]: see https://github.com/NixOS/nixpkgs/blob/master/pkgs/data/themes/matcha/default.nix for an example

**Pull request closed without merging**